### PR TITLE
scio: make scio-test dependency optional

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
Make the `scio-test` dependency `optional`.

## Motivation and Context
We do not want to include scio-test and its dependencies (scalatest etc) in user production artifacts.

To use `scala-test`, users must now add an explicit dependency. That seems to already be common practice.

Note that the `ScioOperator` code still contains references to classes in `scio-test`, but only in methods that are used as part of testing, like `ScioOperator.mock()`.

## Have you tested this? If so, how?
Verified that this works both for running a production scio job and local tests using our internal `flo-example`.

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: contrib/flo-scio_2.11/pom.xml [hunk 1]
```diff
@@ -34,6 +34,7 @@
       <groupId>com.spotify</groupId>
       <artifactId>scio-test_2.11</artifactId>
       <version>0.5.5</version>
```
Comment (honnix) [inline contrib/flo-scio_2.11/pom.xml]: Not part of this PR, but was it on purpose we don't put `version` in parent pom `dependencyManagement`?

Comment (danielnorberg) [reply contrib/flo-scio_2.11/pom.xml]: Probably because that's a scala 2.11 specific dependency.

Comment (honnix) [reply contrib/flo-scio_2.11/pom.xml]: For this https://github.com/spotify/flo/pull/168/files#diff-9b2c165d188171b45f6d1e1dcff5b3f5R26, I usually prefer putting it in parent as well, such as https://github.com/spotify/flo/blob/617987f92362d68baaef38b8d0ec831949458b5d/pom.xml#L60

Comment (danielnorberg) [reply contrib/flo-scio_2.11/pom.xml]: Compare with https://github.com/spotify/flo/blob/master/flo-scala_2.11/pom.xml#L48
